dataset/src/main/kotlin/org/jetbrains/kotlinx/dl/dataset/preprocessor/image/Save.kt

```diff
@@ -24,7 +24,7 @@ public class Save(
     public var dirLocation: File? = null
 ) : ImagePreprocessor {
     @Throws(IOException::class)
-    internal fun imageToFile(filename: String, image: BufferedImage, shape: ImageShape): File {
+    internal fun imageToFile(filename: String, image: BufferedImage): File {
         val outputFile: File = if (dirLocation!!.isDirectory) {
             File("${dirLocation}\\${filename}")
```

Comment:
How about fix this line `File.separator`? This is not correct for unix systems now.

Reply:
Many thanks for the comment, do you have any suggestions for quick fix?

Reply:
I think there is a problem with declaring dirLocation as nullable. Passing null will just throw a NullPointerException. And I want to note that if the value of `dirLocation` is not a directory, then each processing result will overwrite the file.

Reply:
> How about fix this line with File.separator? This is not correct for unix systems now.

I think this fix should be done as a separate pull request.

Reply:
@YokiToki we are going to split into two parts the logic of loading the whole directory and one file, and thanks for the note about the Unix system and possible overwrinting